w3/streaming-multihash/src/hasher.js

```diff
@@ -0,0 +1,56 @@
+import * as API from './type.js'
+import * as Digest from 'multiformats/hashes/digest'
+
+/**
+ * @template {string} Name
+ * @template {number} Code
+ * @param {object} options
+ * @param {Name} options.name
+ * @param {Code} options.code
+ * @param {{create: () => API.RawHasher}} options.hasher
+ */
+
+export const derive = ({ name, code, hasher }) => {
+  /**
+   * @returns {API.StreamingHasher<Code>}
+   */
+  const create = () => new StreamingHasher(code, hasher.create())
```

Comment:
if you agree with all my suggestions, you dont actually need this you can just return the instance directly